packages/kbn-content-management/table_list/src/table_list_view/components/listing_limit_warning.tsx

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+/* eslint react-hooks/exhaustive-deps: 2 */
```

Comment:
Why is this necessary?  And it's hard to know what it applies to as a file-level override.

@spalger What's @elastic/kibana-operations view on this kind of rule override?

Reply:
This sort of "turn on a rule" stuff should really be in the root `.eslintrc.js` file with an `overrides` block selecting the files where this rule applies. We don't have any validation to force this but we could if we need it. We are currently using it in 15 files so it's not super wide-spread but it would make the rule context really hard to understand if it become a widely used practice.

Reply:
Sorry left over from `kibana_react`. That plugin has the rule disabled and I did want to have it enabled on this file.